seat.h

```diff
@@ -14,12 +14,23 @@
 #define DEFAULT_XCURSOR "left_ptr"
 #define XCURSOR_SIZE 24
 
+struct cg_keyboard_group {
+	struct cg_keyboard *keyboard;
+
+	struct wlr_keyboard_group *wlr_group;
+	struct cg_seat *seat;
+	struct wl_listener key;
+	struct wl_listener modifiers;
+	struct wl_list link;
+};
+
 struct cg_seat {
 	struct wlr_seat *seat;
 	struct cg_server *server;
 	struct wl_listener destroy;
 
 	struct wl_list keyboards;
```

Comment:
Yes, I believe it is, since we are still in charge of keeping track of all of the separate keyboards and destroying them when they are removed (when `handle_keyboard_destroy` is called)

Reply:
For the destroy signal we just need the listener, the keyboards don't actually need to be in a list in the seat. The only place where this list is currently used that I'm not sure if it can be replaced with the keyboard group list is the following:

```
wl_list_for_each_safe(keyboard, keyboard_tmp, &seat->keyboards, link) {
	handle_keyboard_destroy(&keyboard->destroy, NULL);
}
```

Is is possible to iterate over the keyboard groups and destroy all keyboards that are in the groups?

Reply:
keyboard groups only have a concept of `wlr_keyboard`s. So in order to make this have even a chance of working, we would have to pitch out the entire `cg_keyboard` structure. We could then iterate over the group's devices, get the `wlr_keyboard` from them, remove them from the group and then destroy them. Do you see any reasons for this not to work? Would we then have to make a list of listeners or is it enough that they are just around?

Reply:
Sorry, I meant we could move the list of `cg_keyboard`s to the `cg_keyboard_group`s.

Hm.. interesting thought, we could try to get rid of `cg_keyboard` and only deal with `cg_keyboard_group`... Are you up for trying? :D

Reply:
Resolving this in favour of creating a new review for the commit which attempts to address this.